### PR TITLE
✨ Improve the Status Controller

### LINKED ISSUE #1862: feature: Clean up the status controller
### Feature Description

We had some discussion about the status controller and all agreed that it can be simplified by doing some cleaning up. Detailed [here](https://github.com/kubestellar/kubestellar/pull/1858#discussion_r1513714628).

### Proposed Solution

Do it.

### Want to contribute?

- [ ] I would like to work on this issue.

### Additional Context

_No response_

### PR DESCRIPTION
**Pull request by waltforme**
## Summary
This PR improves the status controller in two aspects.
1. Idiomatic synchronization between the status controller and the binding controller using a channel. So that the status controller doesn't need to check everyone of the per-resource informers for cache sync. Instead it proceeds immediately when the binding controller knows that all caches are synced.
2. Removal of watch to bindingpolicies.

## Related issue(s)
Closes #1862

### CODE REVIEW

File: pkg/status/controller.go [hunk 1]
```diff
@@ -108,20 +104,18 @@ func NewController(wdsRestConfig *rest.Config, imbsRestConfig *rest.Config,
 		imbsDynClient:  imbsDynClient,
 		imbsKubeClient: imbsKubeClient,
 		workqueue:      workqueue.NewRateLimitingQueue(ratelimiter),
-		listers:        listers,
-		informers:      informers,
```
Comment (andreyod) [inline pkg/status/controller.go:112]: The parameters `listers` and `informers` are no longer used in this func and should be removed from the signature. Right?

Comment (waltforme) [reply pkg/status/controller.go:112]: Exactly. Removed in latest push.

**Pull request merged**